.circleci/config.yml

```diff
@@ -0,0 +1,73 @@
+version: 2
+jobs:
+  build:
+    working_directory: ~/aerogear
+    docker:
+      # Node 8 LTS
+      - image: circleci/node:carbon
+    steps:
+      - checkout
+      # Intentionally leaving out the caching bits first to look at build times
+      # Without caching
+      # - restore_cache:
+      #     key: dependency-cache-{{ checksum "package.json" }}
+      - run:
+          name: install dependencies
+          command: npm install
+      - run:
+          name: bootstrap project
+          command: npm run bootstrap
+      - run:
+          name: compile
+          command: npm run compile
+      # - save_cache:
+      #     key: dependency-cache-{{ checksum "package.json" }}
+      #     paths:
+      #       - ./node_modules
+      # - run:
+      #     # Hoisting will cache some of the packages, but rest needs to be installed separtely
+      #     name: install-package-dependencies-after-cache
+      #     command: npm run bootstrap
+      - run:
+          name: run lint
+          command: npm run lint
+      - run:
+          name: test
+          command: npm test
+      # - store_artifacts:
+      #     path: test-results.xml
+      #     prefix: tests
+      # - store_artifacts:
+      #     path: coverage
+      #     prefix: coverage
+      # - store_test_results:
+      #     path: test-results.xml
+  npm_publish:
+    working_directory: ~/aerogear
+    docker:
+      # Node 8 LTS
+      - image: circleci/node:carbon
+    steps:
+      - checkout
+        # Allows us to authenticate with the npm registry
+      - run: echo "//registry.npmjs.org/:_authToken=$NPM_TOKEN" > ~/.npmrc
+      - run: CI=true npm run release:prep
+      - run: TAG=$CIRCLE_TAG npm run release:validate
+      - run: npm run publish
+
+workflows:
+  version: 2
+  build_and_release:
+    jobs:
+      - build:
+          filters:
+            tags:
+              only: /.*/
+      - npm_publish:
```

Comment:
We can reuse scripts done js-sdk to support this case. Not sure if we want to copy this along with docs now or it will be separate ticket

Reply:
Separate ticket here: https://issues.jboss.org/browse/AEROGEAR-8181